### PR TITLE
MAINTAINER: maintainer file updates

### PR DESCRIPTION
**Pull request by nashif**
- remove andrewboie
- added SiLabs and Nouvoton platforms
- Misc optimizations

### CODE REVIEW

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -228,8 +227,9 @@ CMSIS API layer:
 
 Common arch:
     status: maintained
-    maintainers:
-        - andrewboie
+    collaborators:
```
Comment (ioannisg) [inline MAINTAINERS.yml:230]: So @nashif should the "arch common" area be set to orphaned?

Comment (nashif) [reply MAINTAINERS.yml:230]: I do not think we should call this orphaned, this is such a bad status to have and gives a very bad impression, we already talked about this and the need to remove it completely.

Comment (ioannisg) [reply MAINTAINERS.yml:230]: Well if @andrewboie cannot maintain it, we should elevate one collaborator to maintainer level, then. MAybe ask for volunteers?

Comment (nashif) [reply MAINTAINERS.yml:230]: right now I just want to remove andrew and do not worry about setting maintainers and requiring approval and such, thats why I left maintainers out so this change can go in ASAP. that does not mean the area is orphaned, we have many potential maintainers and owners, we can do that in a separate PR.

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -1199,6 +1204,34 @@ Synopsys Platforms:
...
+        - sjg20
+        - jackrosenthal
+    files:
+        - soc/arm/nuvoton/
+        - boards/arm/npcx*/
+        - dts/arm/nuvoton/
+        - dts/bindings/*/nuvoton/*
+        - drivers/*/*_npcx*
+    labels:
+        - "platform: SiLabs"
```
Comment (chrta) [inline MAINTAINERS.yml:1220]: `SiLabs` -> `Nuvoton`

**Review by chrta — CHANGES_REQUESTED** [review]

**Review by chrta — APPROVED** [review]
LGTM for SiLabs addition

**Review by ioannisg — APPROVED** [review]
Fine with me.

**Pull request merged**